### PR TITLE
Initial prep work for EKS

### PR DESCRIPTION
**Pull request by alexbakervr**
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

## What

Update Dockerfile and puma.rb file in preparation for EKS migration. Add GitHub actions workflow to create ECR repo and push image.

## Why

To prepare the app for migration to EKS.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,44 @@
+ARG base_image=ruby:2.7.2
```
Comment (KludgeKML) [inline Dockerfile:1]: Current ruby version in main is now 2.7.5, so you might want to change this to:
`ARG base_image=ruby:2.7.5-slim-buster`

File: .ruby-version [hunk 2]
```diff
@@ -1 +1 @@
-2.7.2
+2.7.5
```
Comment (fredericfran-gds) [inline .ruby-version:1]: I would keep it on 2.7.2 and let the developers move it later to 2.7.5

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+FROM $base_image AS builder
+
+ENV RAILS_ENV=production
+# TODO: have a separate build image which already contains the build-only deps.
+RUN apt-get update -qq && \ 
+    apt-get upgrade -y && \
+    apt-get install -y build-essential nodejs && \
+    apt-get clean
+
+RUN bundle config set force_ruby_platform true
```
Comment (fredericfran-gds) [inline Dockerfile:12]: maybe add this in line 20 `RUN`

File: .ruby-version (hunk 2 above)
Comment (fredericfran-gds) [reply .ruby-version:1]: Weird your PR is not picking up the change that it was moved to 2.7.5 by the developers 7 days ago. Can you rebase on main, thanks?

**Pull request closed without merging**